### ISSUE TITLE
Assertion `newEdgeNodes.count(std::make_pair(reverse(pos), reverse(before))) == 1 || newEdgeNodes.count(std::make_pair(before, pos)) == 1' failed.

### ISSUE THREAD

**Issue by Tierhon** [marbl/verkko#63]
Hi,
I'm running with MBG 1.0.10, but with verkko 1.0 beta 2 since the 1.0 release haven't been updated in Conda yet. 

```
removed 1 tips
try resolve k=1601, replaced 2 nodes with 5 nodes, unitigified 8 nodes to 4 nodes
try resolve k=1602, replaced 6 nodes with 17 nodes, unitigified 26 nodes to 9 nodes
try resolve k=1603, replaced 7 nodes with 23 nodes, unitigified 36 nodes to 16 nodes
try resolve k=1604, replaced 10 nodes with 26 nodes, unitigified 32 nodes to 16 nodes
try resolve k=1605, replaced 2 nodes with 6 nodes, unitigified 8 nodes to 4 nodes
try resolve k=1606, replaced 5 nodes with 12 nodes, unitigified 14 nodes to 7 nodes
try resolve k=1607, replaced 3 nodes with 9 nodes, unitigified 12 nodes to 6 nodes
try resolve k=1608, replaced 3 nodes with 10 nodes, unitigified 15 nodes to 5 nodes
try resolve k=1609, replaced 3 nodes with 10 nodes, unitigified 16 nodes to 6 nodes
try resolve k=1610, replaced 5 nodes with 15 nodes, unitigified 24 nodes to 10 nodes
try resolve k=1611, replaced 3 nodes with 7 nodes, unitigified 10 nodes to 5 nodes
try resolve k=1612, replaced 7 nodes with 17 nodes, unitigified 24 nodes to 12 nodes
try resolve k=1613MBG: src/UnitigResolver.cpp:1816: ResolutionResult resolve(ResolvableUnitigGraph&, const HashList&, size_t, std::vector<PathGroup>&, const phmap::flat_hash_set<long unsigned int>&, size_t, bool): Assertion `newEdgeNodes.count(std::make_pair(reverse(pos), reverse(before))) == 1 || newEdgeNodes.count(std::make_pair(before, pos)) == 1' failed.
./buildGraph.sh: line 39: 26304 Aborted                 (core dumped) /cluster/projects/nn8013k/programs/miniconda3/envs/verkko/bin/MBG $iopt -t 4 -k 1001 -r 15000 -R 4000 -w 100 --kmer-abundance 1 --unitig-abundance 2 --error-masking=collapse-msat --output-sequence-paths ../1-buildGraph/paths.gaf --out ../1-buildGraph/hifi-resolved.gfa
```
I got the same issue with 1.0.9, but wanted to test with 1.0.10 before reporting. 

This is for an Atlantic salmon. Trying with only HiFi reads so far. I could share data if you want (55 GB uncompressed corrected data).

Thank you.

Ole

**Comment by Tierhon** [comment]
Hi,
I can confirm that the issue is still in the 1.0 release of verkko, not that I expected that to change anything really. 

Should I rather report this on the MBG issue page? 

Thank you.

Ole

**Comment by skoren** [comment]
There's a fix in the tip of MBG: https://github.com/maickrau/MBG/commit/de2b7e859e9710259f199830ec6a18643d1435ae however it isn't tested yet. You could try building MBG from source and seeing if that fix addresses your crash or if it still crashes with an error.

**Comment by Tierhon** [comment]
Great Sergey, I'll try that and report back.

**Comment by Tierhon** [comment]
The tip of MBG fixed the issue I had, but another came up:
```
140428 unitigs after resolving
Building unitig sequences
Reading sequences from ../hifi-corrected.fasta
Writing graph to ../1-buildGraph/hifi-resolved.gfa
Writing paths to ../1-buildGraph/paths.gaf
MBG: src/MBG.cpp:99: void updatePathRemaining(size_t&, size_t&, bool, const DumbSelect&, size_t): Assertion `expandedPoses.countOnes() >= overlap + 2' failed.
./buildGraph.sh: line 39: 62925 Aborted                 (core dumped) /cluster/projects/nn8013k/opt/MBG/bin/MBG $iopt -t 4 -k 1001 -r 15000 -R 4000 -w 100 --kmer-abundance 1 --unitig-abundance 2 --error-masking=collapse-msat --output-sequence-paths ../1-buildGraph/paths.gaf --out ../1-buildGraph/hifi-resolved.gfa
```

Thank you. 

Ole

**Comment by maickrau** [comment]
Hi Ole,

Could you please share `hifi-corrected.fasta`, `1-buildGraph/buildGraph.err` and `1-buildGraph/buildGraph.sh` from the assembly folder

**Comment by Tierhon** [comment]
Hi Mikko,
please download the files from these links:
https://www.dropbox.com/s/ttjinjcobqf14ho/buildGraph.err?dl=0
https://www.dropbox.com/s/vqcusnajx8ar24m/buildGraph.sh?dl=0
https://www.dropbox.com/s/ufolwc9lsltl94u/hifi-corrected.fasta.gz?dl=0

Thank you.

Ole

**Comment by maickrau** [comment]
This should now be fixed in MBG commit [ad2934f](https://github.com/maickrau/MBG/commit/ad2934f0dacaaf7ad21e3c0bd46da67f8a4e85ed), could you please try again with that
